tools/autograd/gen_python_functions.py

```diff
@@ -424,6 +482,16 @@ def get_python_signature(declaration, include_out):
             default = 'None'
         if default is not None:
             param += '=' + str(default)
+        return param
+
+    for arg in declaration['arguments']:
+        if arg.get('output', False):
+            output_args.append(arg)
+            continue
+        if arg.get('kwarg_only', False) and positional:
```

Comment:
There seems like some repeated code handling `*` that could be refactored into a function

Reply:
I'd rather avoid introducing a side-effect to save ~4 lines.